src/views/ManageList.jsx

```diff
@@ -1,7 +1,126 @@
-export function ManageList() {
+import { useState } from 'react';
+import { addItem } from '../api/firebase';
+import toast, { Toaster } from 'react-hot-toast';
+
+const purchaseTimelines = {
+	soon: 7,
+	kindOfSoon: 14,
+	notSoon: 30,
+};
+
+export function ManageList({ listPath }) {
+	const [itemName, setItemName] = useState('');
+	const [itemNextPurchaseTimeline, setItemNextPurchaseTimeline] = useState('');
+
+	const handleItemNameTextChange = (e) => {
+		setItemName(e.target.value);
+	};
+
+	const handleNextPurchaseChange = (e) => {
+		setItemNextPurchaseTimeline(e.target.value);
+	};
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+
+		const daysUntilNextPurchase = purchaseTimelines[itemNextPurchaseTimeline];
+
+		try {
+			await toast.promise(
+				addItem(listPath, {
+					itemName,
+					daysUntilNextPurchase,
+				}),
+				{
+					pending: 'Adding item to list.',
+					success: () => {
+						setItemName('');
+						setItemNextPurchaseTimeline('');
+						return `${itemName} successfully added to your list!`;
+					},
+					error: () => {
+						return `${itemName} failed to add to your list. Please try again!`;
+					},
+				},
+				{
+					style: {
+						minWidth: '250px',
+					},
+				},
```

Comment:
This might potentially be a nit pick suggestion, but is there a way for the style element to not be present in the jsx file, just so styling elements are all saved separately?

Reply:
If I had to take a full guess probably 😅, this was following documentation from the `react-hot-toast` how to set it up with the options. So it would just be a double check on if this is specific to the use of the toast or just an option in the usage. 🤔